src/lib/stable/update.ts

```diff
@@ -31,15 +31,18 @@ export function update(
 
         descriptor.value = methodCallback as any;
 
+        const index = globalThis._azleCanisterMethodsIndex++;
```

Comment:
Wasm uses i32 or i64, and it's simple for our decorators/canister method function things (query, update, init etc) to just increment an index to be assigned to each function. This makes it simple to modify the Wasm binary to export a function that simply calls into an index. If we don't do this it will be relatively complicated to somehow store string names in wasm, as we'd have to do data section/memory manipulation stuff instead of just using simple numbers. This is also how the DFINITY engineer who helped me to design this did it.